Project.toml

```diff
@@ -4,12 +4,14 @@ authors = ["Erik Faulhaber <erik.faulhaber@uni-koeln.de>"]
 version = "0.3.2-dev"
 
 [deps]
+Atomix = "a9b6321e-bd34-4604-b9c9-b65b8de01458"
```

Comment:
Are you sure about `Atomix`? Seems to be dead.
You could just copy the reevant macros from sugar.jl https://github.com/JuliaConcurrent/Atomix.jl/blob/main/src/sugar.jl
Also there seem to be issues on GPUs?

Reply:
It's not dead, it just didn't have to be modified in the past 2 years because it's a simple package.
KernelAbstractions.jl is also based on it:
https://github.com/JuliaGPU/KernelAbstractions.jl/blob/4285051c3ea949aee8a024383f38c8849f602729/src/KernelAbstractions.jl#L12